packages/api/src/routers/tx/getByHash.ts

```diff
@@ -65,6 +67,29 @@ export const getByHash = publicProcedure
         );
       }
 
-      return serializeTransaction(addDerivedFieldsToTransaction(queriedTx));
+      const tx = serializeTransaction(addDerivedFieldsToTransaction(queriedTx));
+
+      if (tx.decodedFields) {
+        tx.decodedFields = await addFullBlockHash(tx.decodedFields);
+      }
+
+      return tx;
     }
   );
+
+async function addFullBlockHash(decoded: string): Promise<string> {
```

Comment:
Maybe it would make more sense to perform the block hash recovering logic inside of the decoding logic itself.

Reply:
Done [0a276e8](https://github.com/Blobscan/blobscan/pull/608/commits/0a276e85baae58d5b9b55b71a167be9369743415)